skimage/filters/tests/test_edges.py

```diff
@@ -364,39 +364,138 @@ def test_laplace_mask():
     assert (np.all(result == 0))
 
 
-@testing.parametrize("grad_func", (
-    filters.prewitt_h, filters.sobel_h, filters.scharr_h
-))
+def test_farid_zeros():
+    """Farid on an array of all zeros."""
+    result = filters.farid(np.zeros((10, 10)), np.ones((10, 10), bool))
+    assert (np.all(result == 0))
+
+
+def test_farid_mask():
+    """Farid on a masked array should be zero."""
+    np.random.seed(0)
+    result = filters.farid(np.random.uniform(size=(10, 10)),
+                           np.zeros((10, 10), bool))
+    assert (np.all(result == 0))
+
+
+def test_farid_horizontal():
+    """Farid on a horizontal edge should be a horizontal line."""
+    i, j = np.mgrid[-5:6, -5:6]
+    image = (i >= 0).astype(float)
+    result = filters.farid(image) * np.sqrt(2)
+    # Fudge the eroded points
+    i[np.abs(j) == 5] = 10000
+    assert (np.all(result[i == 0] == result[i == 0][0]))
+    assert_allclose(result[np.abs(i) > 2], 0, atol=1e-10)
+
+
+def test_farid_vertical():
```

Comment:
In general, no explanations are given for the tests.  What are they testing for and why?

Reply:
Several minor conditions: if the filter returns zeros when applied to a null image, their response on each direction, or how they behave when a particular array is used.